google-cloud-logging/src/main/java/com/google/cloud/logging/Instrumentation.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import com.google.api.client.util.Strings;
+import com.google.api.gax.core.GaxProperties;
+import com.google.cloud.Tuple;
+import com.google.cloud.logging.Payload.JsonPayload;
+import com.google.cloud.logging.Payload.Type;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Instrumentation {
+  public static final String DIAGNOSTIC_INFO_KEY = "logging.googleapis.com/diagnostic";
+  public static final String INSTRUMENTATION_SOURCE_KEY = "instrumentation_source";
+  public static final String INSTRUMENTATION_NAME_KEY = "name";
+  public static final String INSTRUMENTATION_VERSION_KEY = "version";
+  public static final String JAVA_LIBRARY_NAME_PREFIX = "java";
+  public static final String DEFAULT_INSTRUMENTATION_VERSION = "UNKNOWN";
+  public static final int MAX_DIAGNOSTIC_VALUE_LENGTH = 14;
+  public static final int MAX_DIAGNOSTIC_ENTIES = 3;
+  private static boolean instrumentationAdded = false;
+
+  public static Tuple<Boolean, Iterable<LogEntry>> populateInstrumentationInfo(
```

Comment:
nit: consider declaring a class instead of  the `Tuple<?,?>`. The `x()` and `y()` calls in the `LoggingHandler.java` cannot be guessed without reading through all code. Method documentation can be added to elaborate about the returned values.

Reply:
Thanks for input! I like the tuple definition since it does not hides the values it carries and also this code is not on the main path of exposed APIs. Thats said I went ahead and documented the API signature

Reply:
I think by _hiding_ you reference the visual hints that IDE (such as VSCode) expose when working with tuple objects. While it is true that you get hints about the type, it cannot replace the additional info that is conveyed via the good variable/function/class name. Leave it at your discretion though.

Reply:
The variable names could be indeed informative, but creating a class just for grouping returned values could be a bit too much - thats what is `Tuple` designed for. Given a fact that this functionality is not supposed to be used publicly by developers, I think `Tuple` is exactly what we need - I wish we could have an anonymous types or anything similar in Java